src/views/serviceReport/ServiceReport.jsx

```diff
@@ -147,9 +158,40 @@ class ServiceReport extends Component {
     });
     doc.save('Service Report' + '.pdf');
   }
+  onClickFilter(e) {
+    console.error(this.state.startDate, this.state.endDate);
+    this.setState({ filter: true });
+  }
+  sdateChange(date) {
+    this.setState({
+      startDate: date
+    });
+  }
+  edateChange(date) {
+    this.setState({
+      endDate: date
+    });
+  }
+  resetFilter(e) {
+    e.preventDefault();
+    this.setState({
+      endDate: '',
+      startDate: '',
+      filter: false
+    });
+  }
   render() {
     const { Services = [] } = this.props;
     const MyLoader = () => <Loader loading={Services.loading} />;
```

Comment:
initialise tableloader in helper function and import here. 
pls avoid component initialising in render method